File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/Send_chat_message_024_Secure.py
messages = []


VALID_USERS = {
    "user_1",
    "user_2",
    "user_3"
}


def send_chat_message(sender_id, recipient_id, message_text):
    """
    Securely sends a chat message.
    """

    try:

        if not sender_id or not recipient_id or not message_text:
            return "Required fields are missing."

        sender_id = sender_id.strip()
        recipient_id = recipient_id.strip()
        message_text = message_text.strip()

        if sender_id not in VALID_USERS:
            return "Invalid sender."

        if recipient_id not in VALID_USERS:
            return "Invalid recipient."

        if not message_text:
            return "Message cannot be empty."

        messages.append({
            "sender_id": sender_id,
            "recipient_id": recipient_id,
            "message": message_text
        })

        return "Message delivered successfully."

    except Exception:
        return "Unable to send message."


if __name__ == "__main__":
    print(send_chat_message(
        "user_1",
        "user_2",
        "Hello!"
    ))

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/Send_chat_message_024_Secure.java
import java.util.ArrayList;
import java.util.List;
import java.util.Set;

public class Send_chat_message_024_Secure {

    static class Message {
        String senderId;
        String recipientId;
        String text;

        Message(String senderId, String recipientId, String text) {
            this.senderId = senderId;
            this.recipientId = recipientId;
            this.text = text;
        }
    }

    static List<Message> messages = new ArrayList<>();

    static Set<String> users = Set.of(
            "U100",
            "U200",
            "U300"
    );

    public static String sendMessage(
            String senderId,
            String recipientId,
            String message) {

        if (senderId == null || recipientId == null || message == null)
            return "Required fields are missing.";

        senderId = senderId.trim();
        recipientId = recipientId.trim();
        message = message.trim();

        if (senderId.isEmpty() ||
            recipientId.isEmpty() ||
            message.isEmpty())
            return "Required fields are missing.";

        if (!users.contains(senderId))
            return "Sender does not exist.";

        if (!users.contains(recipientId))
            return "Recipient does not exist.";

        messages.add(new Message(senderId, recipientId, message));

        return "Message delivered successfully.";
    }

    public static void main(String[] args) {

        System.out.println(sendMessage(
                "U100",
                "U200",
                "Hello!"
        ));
    }
}